cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -173,48 +183,35 @@ func (csr *ClusterStateRegistry) updateScaleRequests(currentTime time.Time) {
 	// clean up stale backoff info
 	csr.nodeGroupBackoffInfo.RemoveStaleBackoffData(currentTime)
 
-	timedOutSur := make([]*ScaleUpRequest, 0)
-	newSur := make([]*ScaleUpRequest, 0)
-	for _, sur := range csr.scaleUpRequests {
-		if !csr.areThereUpcomingNodesInNodeGroup(sur.NodeGroupName) {
+	for nodeGroupName, scaleUpRequest := range csr.scaleUpRequests {
+		if !csr.areThereUpcomingNodesInNodeGroup(nodeGroupName) {
 			// scale-out finished successfully
 			// remove it and reset node group backoff
-			csr.nodeGroupBackoffInfo.RemoveBackoff(sur.NodeGroupName)
+			csr.nodeGroupBackoffInfo.RemoveBackoff(nodeGroupName)
 			glog.V(4).Infof("Scale up in group %v finished successfully in %v",
-				sur.NodeGroupName, currentTime.Sub(sur.Time))
+				nodeGroupName, currentTime.Sub(scaleUpRequest.Time))
 			continue
 		}
-		if sur.ExpectedAddTime.After(currentTime) {
-			newSur = append(newSur, sur)
-		} else {
-			timedOutSur = append(timedOutSur, sur)
-		}
-	}
-	csr.scaleUpRequests = newSur
-	for _, sur := range timedOutSur {
-		// IsNodeGroupScalingUp returns true if there is another
-		// scale-up still going on for this group, so it's ok for node
-		// group to still have upcoming nodes. If there is no other
-		// scale-up we have VMs that failed to provision within timeout,
-		// so we consider it a failed scale-up
-		if !csr.IsNodeGroupScalingUp(sur.NodeGroupName) {
+
+		if scaleUpRequest.ExpectedAddTime.Before(currentTime) {
```

Comment:
I know it's terribly, terribly unlikely, but by using Before() here and After() later, we'll quietly drop requests if ExpectedAddTime == currentTime

Reply:
Good catch! Will fix.